var/spack/repos/builtin/packages/libvterm/package.py

```diff
@@ -25,6 +25,9 @@ class Libvterm(Package):
 
     depends_on("libtool", type="build")
 
-    def install(self, spec, prefix):
-        make()
-        make("install", "PREFIX=" + prefix)
+    def setup_build_environment(self, env):
+        env.set("LIBTOOL", self.spec["libtool"].prefix.bin.join("libtool"))
+
+    def edit(self, spec, prefix):
+        makefile = FileFilter("Makefile")
```

Comment:
I wouldn't modify files when the alternative is setting variables on the command line

Reply:
I'm not a Makefile expert...but AFAIK `PREFIX=` does not support setting it externally. Isn't it? Shouldn't it use `PREFIX?=`?

Reply:
```Makefile
$ cat Makefile 
A ?= A
B = B
C := C
override D = D
override E := E
$(info $(A) $(B) $(C) $(D) $(E))
```

``shell
$ make A=x B=x C=x D=x E=x
x x x D E
```

anything that doesn't have `override` can be set on the command line, `?=` is special in that it initializes from env variables too

Reply:
TIL! Thanks for the explanation.

However, I would have to customise both `build` and `install` phases, in order to pass the correct prefix to both (otherwise I might have "misaligned" phases). On the contrary by "fixing" the Makefile, I set it only once and any other call to make will have the correct value.

I'm not sure if it is needed by both, or it can be enough passing PREFIX to `install` (probably PREFIX is not used in `build` phase)...but that sounds less clearer to me, indeed it requires a bit of knowledge on the specific package Makefile content.

Let me know if it makes sense and if you have any preference 😉

Reply:
Right, personally I would only use `def edit` + filter file when a makefile has smth hard-coded that cannot be overridden. Or some Makefiles have includes for a file with variables, that's also a good use for `def edit`.

But in general I would not use `def edit` cause it's not like `patch`: potentially it's executed multiple times in `spack develop` / `dev-build`, so you would have to be sure that it is idempotent. (Which it is in this case as long as you don't have new lines in your path (which is allowed on linux :p))

Reply:
Good point!

Have a look at https://github.com/spack/spack/pull/34776/commits/3773ce3c1f96fb6383eee632fc24e51699dd3074 😉